ansible/roles/snactor/tasks/main.yml

```diff
@@ -0,0 +1,50 @@
+---
+- name: Install dnf-plugins-core to enable using copr repositories
+  become: true
+  dnf:
+      name: dnf-plugins-core
+      state: present
+
+- name: Enable leapp repository
+  become: true
+  raw: "dnf copr enable -y {{ snactor.copr.repo }}"
+
+- name: Get snactor build dependencies
+  raw: "dnf repoquery -C --arch src --requires {{ snactor.package }} 2> /dev/null"
+  register: builddeps
+
+- name: Install packages required while building snactor, if necessary
+  become: true
+  dnf:
+      name: "{{ item }}"
+      state: present
+  with_items: "{{ builddeps.stdout_lines }}"
+
+- name: Get snactor dependencies
+  raw: "dnf repoquery -C --requires {{ snactor.package }} 2> /dev/null"
+  register: pkgdeps
+
+- name: Install packages required by snactor, if necessary
+  become: true
+  dnf:
```

Comment:
did you test that with centos?

Reply:
Unfortunately it will not work, we will have to use yum in Centos7/RHEL7. 
@artmello: I would suggest to check the facts in order to detect what distribution we have and then execute specific module (blocks will work with this well). Or dirty way using command executing dnf -h and then capturing the state. If it will be successful then you can proceed with dnf module, else yum module. But the second way is really dirty and I prefer using facts.

P.S. I just found this in blocks description:  http://docs.ansible.com/ansible/latest/playbooks_blocks.html (see the first example)